contracts/vaults/bStablestMATICVault.sol

```diff
@@ -0,0 +1,165 @@
+
+import "../interfaces/IVault.sol";
+import "@openzeppelin-4.7.0/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin-4.7.0/token/ERC20/ERC20.sol";
+import "../interfaces/IBalancerGauge.sol";
+import "../interfaces/IBalancerVault.sol";
+import "../interfaces/IBalancerPool.sol";
+import "../../interfaces/IPriceFeeds.sol";
+contract bStablestMATICVault is ERC20, IVault {
+    using SafeERC20 for IERC20;
+    address public constant asset = 0xaF5E0B5425dE1F5a630A8cB5AA9D97B8141C908D;
+
+    uint256 public totalAssets;
+
+    address internal constant _bStableGauge = 0x9928340f9E1aaAd7dF1D95E27bd9A5c715202a56;
+    address internal constant _vault = 0xBA12222222228d8Ba445958a75a0704d566BF2C8;
+
+    address public constant BAL = 0x9a71012B13CA4d3D0Cdc72A177DF3ef03b0E76A3;
+
+    uint256 internal _sharePrice = 1e18;
+
+    bytes32 public constant poolId = 0xaf5e0b5425de1f5a630a8cb5aa9d97b8141c908d000200000000000000000366;
+    address public constant WMATIC = 0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270;
+    address public constant STMATIC = 0x3A58a54C066FdC0f2D55FC9C89F0415C92eBf3C4;
+
+    bytes32 public constant poolIdSwap = 0x0297e37f1873d2dab4487aa67cd56b58e2f27875000100000000000000000002;
+
+    address public constant priceFeed = 0x600F8E7B10CF6DA18871Ff79e4A61B13caCEd9BC;
+
+    constructor () ERC20("bStable-stMATIC/MATIC-Vault", "OVault") {}
+
+    function convertToShares(uint256 assets) public view returns (uint256 shares) {
+        return assets*1e18/_sharePrice;
+    }
+
+    function convertToAssets(uint256 shares) public view returns (uint256 assets) {
+        return shares*_sharePrice/1e18;
+    }
+
+    function maxDeposit(address receiver) external view returns (uint256) {
+        return type(uint256).max;
+    }
+
+    //Note: Due to how compounding works with Balancer, the share amount will likely be overstated; however, there is no loss of funds as the share price will increase with it
+    function previewDeposit(uint256 assets) public view returns (uint256) {
+        return convertToShares(assets);
+    }
+
+    function deposit(uint256 assets, address receiver) external returns (uint256 shares) {
+        compound();
+        IERC20(asset).safeTransferFrom(msg.sender, address(this), assets);
+        shares = convertToShares(assets);
+        IBalancerGauge(_bStableGauge).deposit(assets);
+        _mint(receiver, shares);
+        emit Deposit(msg.sender, receiver, assets, shares);
+    }
+
+    function maxMint(address receiver) external view returns (uint256) {
+        return type(uint256).max;
+    }
+
+    //Note: Due to how compounding works with Balancer, the asset amount will likely be understated; however, there is no loss of funds as the share price will increase with it
+    function previewMint(uint256 shares) external view returns (uint256 assets) {
+        return convertToAssets(shares);
+    }
+
+    function mint(uint256 shares, address receiver) external returns (uint256 assets) {
```

Comment:
its either mint/burn or
deposit/withdraw

Reply:
no its both. The vault is ERC-4626 compliant

Reply:
oh ok.